content/announcing-official-typescript-types-public-preview.md

```diff
@@ -0,0 +1,417 @@
+
+- `import Transition from '@ember/routing/-private/transition'` → `import Transition from '@ember/routing/transition'`
+- `import RouteInfo from '@ember/routing/-private/route-info'` → `import RouteInfo from '@ember/routing/route-info'`
+- `import { RouteInfoWithMetadata } from '@ember/routing/-private/route-info-with-metadata'` → `import { RouteInfoWithMetadata } from '@ember/routing/route-info'`
+
+
+## New TypeScript users
+
+*This section only applies if you are trying out the types for the first time!*
+
+For the moment, the best way to get started with these types is to install `ember-cli-typescript` and use its generators, then *remove* a lot of what it does. We will be fixing this in the weeks ahead!
+
+Here’s the process as of today:
+
+1. Set up `ember-cli-typescript` by running `ember install ember-cli-typescript`.
+
+2. Remove the following newly-added packages from your `package.json`:
+
+    - `@types/ember`
+    - `@types/ember__application`
+    - `@types/ember__array`
+    - `@types/ember__component`
+    - `@types/ember__controller`
+    - `@types/ember__debug`
+    - `@types/ember__destroyable`
+    - `@types/ember__engine`
+    - `@types/ember__error`
+    - `@types/ember__helper`
+    - `@types/ember__modifier`
+    - `@types/ember__object`
+    - `@types/ember__owner`
+    - `@types/ember__polyfills`
+    - `@types/ember__routing`
+    - `@types/ember__runloop`
+    - `@types/ember__service`
+    - `@types/ember__string`
+    - `@types/ember__template`
+    - `@types/ember__test`
+    - `@types/ember__utils`
+
+3. Install a version of `ember-source` greater than `4.8.0-beta.2`. You should pick the latest beta of 4.8 *or* any stable version starting with 4.8.0 once it's out!
+
+4. In the `types/<your-app>/index.d.ts` file generated for you automatically by `ember-cli-typescript`, add the new imports as the first items in the file, and remove the array prototype extensions support from the file. With the defaults generated for you, the result would look like this:
+
+    ```diff
+    +import 'ember-source/types';
+    +import 'ember-source/types/preview';
+    -import Ember from 'ember';
+    -
+    -declare global {
+    - // Prevents ESLint from "fixing" this via its auto-fix to turn it into a type
+    - // alias (e.g. after running any Ember CLI generator)
+    - // eslint-disable-next-line @typescript-eslint/no-empty-interface
+    - interface Array<T> extends Ember.ArrayPrototypeExtensions<T> {}
+    - // interface Function extends Ember.FunctionPrototypeExtensions {}
+    -}
+
+    export {};
+    ```
+
+
+## Known issues
+
+The first beta release has the following known issues (which we will fix during the beta period):
+
+- The `@types/ember-data` packages are not compatible with these types, because they assume. If you are using Ember Data with TypeScript, you will need to wait for a future update.
```

Comment:
I think there's a wording issue here.

Reply:
Haha you are correct. Thank you!

Reply:
```suggestion
- The `@types/ember-data` packages are not compatible with these types, because they assume the presence of many of the Ember Classic types we removed in this migration. If you are using Ember Data with TypeScript, you will need to wait for a future update.
```